File: UPCRideAndroid/app/src/main/java/com/example/upcride/activities/RecyclerViewAdaptador.java
package com.example.upcride.activities;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.upcride.R;

import java.util.List;


public class RecyclerViewAdaptador extends RecyclerView.Adapter<RecyclerViewAdaptador.ViewHolder> {

    public static class ViewHolder extends RecyclerView.ViewHolder{

        private TextView nombre, fecha, descripcion, pnt_partida, pnt_destino, n_solis, n_resenias;
        public ImageView foto_perfil;

        public ViewHolder(View itemView){
            super(itemView);
            nombre = (TextView) itemView.findViewById(R.id.n_usuario);
            fecha = (TextView) itemView.findViewById(R.id.d_fecha);
            descripcion = (TextView) itemView.findViewById(R.id.t_descripcion);
            pnt_partida = (TextView) itemView.findViewById(R.id.pnt_partida);
            pnt_destino = (TextView) itemView.findViewById(R.id.pnt_destino);
            n_solis = (TextView) itemView.findViewById(R.id.n_solis);
            n_resenias = (TextView) itemView.findViewById(R.id.n_resenias);
            foto_perfil = (ImageView) itemView.findViewById(R.id.foto_perfil);
        }
    }

    public List<ViajeModelo> viajeLista;

    public RecyclerViewAdaptador(List<ViajeModelo> viajeLista){
        this.viajeLista = viajeLista;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType){
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_viaje, parent, false);
        ViewHolder viewHolder = new ViewHolder(view);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.nombre.setText(viajeLista.get(position).getNombre());
        holder.fecha.setText(viajeLista.get(position).getFecha());
        holder.descripcion.setText(viajeLista.get(position).getDescripcion());
        holder.pnt_partida.setText(viajeLista.get(position).getPnt_partida());
        holder.pnt_destino.setText(viajeLista.get(position).getPnt_destino());
        holder.n_solis.setText(viajeLista.get(position).getN_solis());
        holder.n_resenias.setText(viajeLista.get(position).getN_resenias());
        holder.foto_perfil.setImageResource(viajeLista.get(position).getFoto_perfil());
    }

    @Override
    public int getItemCount() {
        return viajeLista.size();
    }
}

File: UPCRideKotlin/app/src/main/java/com/example/upcridekotlin/activities/RecyclerViewAdaptador.kt
package com.example.upcridekotlin.activities

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.View.OnClickListener
import android.view.ViewGroup
import android.widget.ImageButton
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.upcridekotlin.R
import com.example.upcridekotlin.model.Viaje
import com.example.upcridekotlin.model.ViajeModelo


class RecyclerViewAdaptador(var viajeLista: List<Viaje>) :
    RecyclerView.Adapter<RecyclerViewAdaptador.ViewHolder>(), OnClickListener {


   var  listener : OnClickListener? = null



    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView),OnClickListener {





        var id: Int
        var nombre: TextView
        var fecha: TextView
        var descripcion: TextView
        var pnt_partida: TextView
        var pnt_destino: TextView
        var n_solis: TextView
        var n_resenias: TextView
        var foto_perfil: ImageView
        var btnSolicitar: ImageButton
        var context: Context
        var pasajero : Int

        init {
            id = 0
            pasajero = 0
            nombre = itemView.findViewById<View>(R.id.n_usuario) as TextView
            fecha = itemView.findViewById<View>(R.id.d_fecha) as TextView
            descripcion = itemView.findViewById<View>(R.id.t_descrip) as TextView
            pnt_partida = itemView.findViewById<View>(R.id.pnt_partida) as TextView
            pnt_destino = itemView.findViewById<View>(R.id.pnt_destino) as TextView
            n_solis = itemView.findViewById<View>(R.id.n_solis) as TextView
            n_resenias = itemView.findViewById<View>(R.id.n_resenias) as TextView
            foto_perfil = itemView.findViewById<View>(R.id.foto_perfil) as ImageView
            btnSolicitar = itemView.findViewById<View>(R.id.btnSolicitar) as ImageButton
            context = itemView.context

        }

        fun setOnClickListeners(){
           btnSolicitar.setOnClickListener(this)
        }

        override fun onClick(v: View?) {
            when(v!!.id){
                R.id.btnSolicitar -> {

                    var bundle = Bundle()
                    var intent = Intent(context, solicitar_viaje::class.java)

                    bundle.putInt("idViaje",id)
                    intent.putExtras(bundle)

                    context.startActivity(intent)

                }
            }
        }

    }


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_viaje, parent, false)
        view.setOnClickListener(this)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.id = viajeLista[position].id
        holder.nombre.setText(viajeLista[position].conductor!!.nombres)
        holder.fecha.setText(viajeLista[position].fecha)
        holder.descripcion.setText(viajeLista[position].descripcion)
        holder.pnt_partida.setText(viajeLista[position].puntoPartida)
        holder.pnt_destino.setText(viajeLista[position].puntoDestino)
        holder.n_solis.setText("20")
        holder.n_resenias.setText("20")
        holder.foto_perfil.setImageResource(R.drawable.user)
        holder.setOnClickListeners()
    }

    fun setOnClickListener(listener : OnClickListener )
    {
        this.listener = listener

    }


    override fun onClick(view:View)
    {
        if(listener!=null)

        {
            listener!!.onClick(view)
        }
    }


    override fun getItemCount(): Int {
        return viajeLista.size
    }


}
